### ISSUE TITLE
`turbo:load` event not triggered on first page load and reloads when used with importmap-rails gem.

### ISSUE THREAD

**Issue by edariedl** [hotwired/turbo-rails#216]
Turbo itself is loaded fine, navigation through the application works as expected but for some reason `turbo:load` event is not triggered on first page load and page reloads.

Steps to reproduce:

- Create new rails 6 app
- Add `importmap-rails` and `hotwire-rails` gems (in this order) to your Gemfile and run `bundle`
- Run `rails importmap:install` and `rails hotwire:install`
- Run `bundle` again (to install redis gem inserted by hotwire)
- Create some scaffold eg `rails g scaffold post title author body:text` and run migrations
- add this code to your `app/javascript/application.js` file:
  ```js
  import "@hotwired/stimulus-importmap-autoloader"
  import "@hotwired/turbo-rails"

  document.addEventListener('turbo:load', (_e) =>
    alert("turbo:load triggered")
  )
  ```
- start the rails server `rails s` and go to `localhost:3000/posts` in browser of your choice

Alert won't show on the first page load but it will show on any subsequent page visit after clicking on any link on the page.

**Comment by dhh** [comment]
turbo:load is only fired when turbo drives a new visit. This is by design.

**Comment by edariedl** [comment]
@dhh thank you for your reply. It works this way only with importmap-rails gem. `turbo:load` is fired on the first page load when Turbo is used with webpacker or loaded directly via `turbo_include_tags`. 

Documentation also says it should be fired on initial pageload see https://turbo.hotwired.dev/reference/events
> turbo:load fires once after the initial page load, and again after every Turbo visit. Access visit timing metrics with the event.detail.timing object.

**Comment by dhh** [comment]
Sorry, my confusion. Was thinking about error page loads.

**Comment by dhh** [comment]
Hmm, yeah, replicated this in any other browser than Chrome where we are going through the es-modules-shim.

**Comment by edariedl** [comment]
🙏, sorry I forgot to mention I've tested it in Safari.

**Comment by dhh** [comment]
Filed an issue with es-module-shims: https://github.com/guybedford/es-module-shims/issues/161 

Thanks for finding this!

**Comment by dhh** [comment]
Fixed in importmap-rails 0.3.3 👍
